workers/unity/Assets/Playground/Scripts/Worker/SteamClientWorkerConnector.cs

```diff
@@ -0,0 +1,83 @@
+﻿using Improbable.Gdk.Core;
+// using Steamworks; // Uncomment this line after installing Steamworks.NET. (https://steamworks.github.io)
+using UnityEngine;
+
+namespace Playground
+{
+    /// <summary>
+    ///     The SteamClientWorkerConnector class provides a sample solution for connecting a game client published
+    ///     on Steam to a SpatialOS deployment by retrieving a Steam auth session token at runtime.
+    ///     You can also use the ClientWorkerConnector parent class to parse the Steam auth session token
+    ///     from the commandline arguments in case the Steam auth session token is not known in advance.
+    ///     You will need to install Steamworks.NET (https://steamworks.github.io) in order to use this code.
+    /// </summary>
+    public class SteamClientWorkerConnector : ClientWorkerConnector
+    {
+        protected override bool ShouldUseLocator()
+        {
+            return true;
+        }
+
+        /// <summary>
+        ///     Creates a Locator configuration where the Steam ticket field is populated through an API call.
+        /// </summary>
+        /// <remarks>
+        ///     All other configuration parameters are populated according to the logic of the parent class implementation.
+        ///     (Based on the command line arguments)
+        /// </remarks>
+        /// <param name="workerType">The type of the worker to create.</param>
+        /// <returns>The Locator connection configuration</returns>
+        protected override LocatorConfig GetLocatorConfig(string workerType)
+        {
+            var config = base.GetLocatorConfig(workerType);
```

Comment:
It's possible to pass in a steam ticket through command line arguments and the parent class method would successfully parse and use it in those cases. This class here exists because you Usually don't know the ticket in advance to pass it in as a command line argument. I could slightly change the logic so that we only retrieve a ticket if the command line argument was not specified, i.e. the command line argumet takes priority.

Reply:
I see, that would be nice indeed!

But my comment was about potential error cases, if the command line makes the base class try using a `LoginToken` we'll get an exception saying "can't use both logintoken and steam", but if the base class won't try this it should be fine

Reply:
Oh dear, that some serious OOP spaghetti here. I would propose adding another command line argument called something like `connection_method` with `receptionist, locator_logintoken, locator_steam` being possible values. If we had that we could just ignore superfluous args rather than determining the connection method implicitly. The problem here is that the launcher is still starting game clients with hardcoded commandline args so this change is hard to make. My suggestion would be: let's temporarily accept the current imperfect situation and do this properly once launcher command line args are configurable.

Reply:
Fixed!